### PR TITLE
sucuri parser

### PR DESCRIPTION
**Pull request by dargen3**
There is new sucuri parser. These parser takes data from http collector which takes it from website http://labs.sucuri.net/?malware. Now, it takes only hidden iframes. Domains from website contains non-standart html strings like "&#46". Now parser don't replace them.

### CODE REVIEW

File: intelmq/bots/parsers/sucuri/parser.py [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+        actual_line = report_list[index]
+        while actual_line != "<h4>Hidden iframes</h4>": # displacement to target table
+            index += 1
+            actual_line = report_list[index]
+        while actual_line[:8] != "</tbody>": # scrabing table data
+            index += 1
+            actual_line = report_list[index]
+            if actual_line[:2] == "<t":
+                event = self.new_event(report)
+                event.add("source.url", actual_line.split("\"")[actual_line.count("\"")][1:].split("</a>")[0]) # taking pure data from html line
```
Comment (wagner-certat) [inline intelmq/bots/parsers/sucuri/parser.py]: "http://poseyhumane&#46;org/stats&#46;php" (from the tests) is not a valid URL

File: intelmq/tests/bots/parsers/sucuri/test_sucuri.data [hunk 2]
```diff
@@ -0,0 +1 @@
...dGlvbnM8L2E+PC90ZD48dGQ+PHNwYW4+PGEgaHJlZj0iLz9kZXRhaWxzPWdvbGRwb2xlLmNvbSI+aHR0cDovL2dvbGRwb2xlJiM0NjtjPCEtLSBzdWN1cmkgLS0+b20vP2Y8IS0tIHN1Y3VyaSAtLT5vbGlvPTlQPCEtLXN1Y3VyaS0tPk82WjNNVkY8L2E+PC9zcGFuPjwvdGQ+PC90cj4KPHRyPjx0ZD4yPC90ZD48dGQ+PGEgaHJlZj0iLz90eXBlPXJlZGlyZWN0aW9ucyI+cmVkaXJlY3Rpb25zPC9hPjwvdGQ+PHRkPjxzcGFuPjxhIGhyZWY9Ii8/ZGV0YWlscz1kcmVzc2xpa2UuY29tIj5odHRwOi8vZHJlc3NsaWtlJiM0NjtjPCEtLSBzdWN1cmkgLS0+b20vP2Y8IS0tIHN1Y3VyaSAtLT5vbGlvPTlQPCEtLXN1Y3VyaS0tPk82WjNNVkY8L2E+PC9zcGFuPjwvdGQ+PC90cj4KPC90Ym9keT48dGZvb3Q+PHRyPjx0ZD48L3RkPjx0ZD48L3RkPjx0ZD48aT5MaW1pdGVkIHZpZXcgKDQwIHJvd3MpLi4uIE9ubHkgdGhlIHRvcCBlbnRyaWVzIGJlaW5nIGRpc3BsYXllZC48L2k+PC90ZD48L3RyPjwvdGZvb3Q+PC90YWJsZT48L3A+Cgo8aDQ+RW5jb2RlZCBqYXZhc2NyaXB0PC9oND4KPHA+CkVuY29kZWQgamF2YXNjcmlwdCAocmVkaXJlY3RpbmcgdG8gYmxhY2tob2xlIGFuZCBvdGhlciBleHBsb2l0IGtpdHMpIG9yIHRvIGJ1aWxkCmEgcmVtb3RlIGNhbGwuCjx0YWJsZSBjbGFzcz0iZ3B0YWJsZSBtd3RhYmxlIHRhYmxlIj48dGhlYWQ+PHRyPjx0aD4jIG9mIHNpdGVzIGluZmVjdGVkPC90aD48dGg+VHlwZTwvdGg+PHRoPk1hbHdhcmUgLyBEb21haW5zPC90aD48L3RyPjwvdGhlYWQ+PHRib2R5Pjx0cj48dGQ+MTI8L3RkPjx0ZD48YSBocmVmPSIvP3R5cGU9amF2YXNjcmlwdCI+amF2YXNjcmlwdDwvYT48L3RkPjx0ZD48c3Bhbj48YSBocmVmPSIvP2RldGFpbHM9ZGl2LWNsYXNzLWNvbnRhaW5l
```
Comment (wagner-certat) [inline intelmq/tests/bots/parsers/sucuri/test_sucuri.data]: Please add this file decoded - human readable and only encoded it in the test when it is needed. Makes it a lot easier for humans to inspect the data format.

File: intelmq/tests/bots/parsers/sucuri/test_sucuri.py [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+import unittest
+import intelmq.lib.test as test
+from intelmq.bots.parsers.sucuri.parser import SucuriParserBot
+
+
+with open(os.path.join(os.path.dirname(__file__), 'test_sucuri.data')) as handle:
+    REPORT_DATA = handle.read().strip()
+
+REPORT = {"feed.name": "Sucuri security", "raw": REPORT_DATA, "__type": "Report", "feed.url": "http://labs.sucuri.net/?malware", "time.observation": "2018-01-22T14:38:24+00:00", "feed.accuracy": 100.0}
+EVENT = {'raw': 'PHRhYmxlIGNsYXNzPSJncHRhYmxlIG13dGFibGUgdGFibGUiPjx0aGVhZD48dHI+PHRoPiMgb2Ygc2l0ZXMgaW5mZWN0ZWQ8L3RoPjx0aD5UeXBlPC90aD48dGg+TWFsd2FyZSAvIERvbWFpbnM8L3RoPjwvdHI+PC90aGVhZD48dGJvZHk+PHRyPjx0ZD41NTwvdGQ+PHRkPjxhIGhyZWY9Ii8/dHlwZT1pZnJhbWUiPmlmcmFtZTwvYT48L3RkPjx0ZD48c3Bhbj48YSBocmVmPSIvP2RldGFpbHM9cG9zZXlodW1hbmUub3JnIj5odHRwOi8vcG9zZXlodW1hbmUmIzQ2O29yZy9zdGF0cyYjNDY7cGhwPC9hPjwvc3Bhbj48L3RkPjwvdHI+', 'classification.type': 'blacklist', "feed.name": "Sucuri security", "feed.accuracy": 100.0, "__type": "Event", "feed.url": "http://labs.sucuri.net/?malware", "time.observation": "2018-01-22T14:38:24+00:00", "source.url": "http://poseyhumane&#46;org/stats&#46;php"}
```
Comment (wagner-certat) [inline intelmq/tests/bots/parsers/sucuri/test_sucuri.py]: Where does the blacklist classification come from? I don't see it in the parser.

File: intelmq/tests/bots/parsers/sucuri/test_sucuri.data (hunk 2 above)
Comment (dargen3) [reply intelmq/tests/bots/parsers/sucuri/test_sucuri.data]: I don't understand exactly. Should i have in intelmq/tests/bots/parsers/sucuri/ two data files? First encoded file for test and second decoded file for easier inspekt? Or should i have there only decoded file and encoded data should be in test_sucuri.py?

File: intelmq/tests/bots/parsers/sucuri/test_sucuri.py (hunk 3 above)
Comment (dargen3) [reply intelmq/tests/bots/parsers/sucuri/test_sucuri.py]: I apologize for lot of mistakes. I am absolutely new in intelmq. Now i try to repair it. I have some questions above.

Comment (dargen3) [reply intelmq/tests/bots/parsers/sucuri/test_sucuri.py]: Should have event only classification.identifier = hidded iframes or should have classification.identifier = hidden inframes and classification.type = blacklist?

File: intelmq/tests/bots/parsers/sucuri/test_sucuri.data (hunk 2 above)
Comment (wagner-certat) [reply intelmq/tests/bots/parsers/sucuri/test_sucuri.data]: Just save the plain HTML here. When reading the file you can encode it on the fly, e.g. https://github.com/certtools/intelmq/blob/develop/intelmq/tests/bots/parsers/generic/test_parser_csv.py#L16

File: intelmq/tests/bots/parsers/sucuri/test_sucuri.py (hunk 3 above)
Comment (wagner-certat) [reply intelmq/tests/bots/parsers/sucuri/test_sucuri.py]: Both

**Comment by dargen3** [comment]
Now, there should be all mentioned errors removed.

**Comment by wagner-certat** [comment]
I opened a PR with some additional fixes and enhancements see dargen3/intelmq#1

**Comment by wagner-certat** [comment]
Thanks @dargen3!

**Pull request merged**